engine/components-api/src/main/resources/META-INF/cloudstack/core/spring-engine-components-api-core-context.xml

```diff
@@ -25,6 +25,6 @@
                       http://www.springframework.org/schema/aop http://www.springframework.org/schema/aop/spring-aop.xsd
                       http://www.springframework.org/schema/context
                       http://www.springframework.org/schema/context/spring-context.xsd"
-                      >                     
-
+                      >
+    <bean id="storageUtil" class="com.cloud.storage.StorageUtil" />
```

Comment:
The result is the same, but what about using `@Component` instead?  You are already using the `@Inject` annotaiton.

Reply:
I just copied/pasted and modified this from another place so I could inject it. Can you give me more details as to exactly what you're looking to be changed here? Thanks

Reply:
Ok, let me provide you with some history.

When Spring was first launched we defined all beans in XML; similarly to what you are doing. However, if you take a closer look, you are not declaring the “injection”  of the properties because the properties are annotated with “@Inject” in the class.

If we were to do the creation of “storageUtil” only in XML we would use something like:
```
<bean id="storageUtil" class="com.cloud.storage.StorageUtil" >
     <proerty name=”clusterDao” ref=”clusterDao” />
      <proerty name=”hostDao” ref=”hostDao” />
      <proerty name=”storagePoolDao” ref=”primaryDataStoreDao” />
      <proerty name=”vmInstanceDao” ref=”vmInstanceDao” />
      <proerty name=”volumeDao” ref=”volumeDao” />
</bean>
```

With time things evolved and annotations started to be used. Spring as other frameworks adopted it. The annotation form for the “bean” declaration in XML is “@Component”. There are other alternatives such as the “@Services”, ”@WebComponent”, and others. They are used as markers for beans that play a role in different layers of the application (Frontend layer, service layer, database and so on).

The annotation to mark properties that need to be injected are the “@Inject” and “@Autowired”. In your case, you are not using the “property” element in XML because you applied the “@Inject” annotation in your class properties.

I do prefer to use either everything in XML or everything in annotation form. Therefore, you can simply remove this line here, and annotate the class with “@Component”. Bear in mind that for the annotation to work we need to enable the annotation scan, but I think ACS is already doing that.

Reply:
Thanks for the detailed explanation, @rafaelweingartner! I believe I now have implemented all of your requests.

Reply:
@rafaelweingartner Can you take a look at my most recent commit? It seems my @Component annotation is causing trouble starting the management server. Thanks

Reply:
OK, @rafaelweingartner, I removed the @ Component and put the XML file back to the way it was per your comments. I think I've implemented all of your review comments now. Thanks!